mongo/integration/sdam_error_handling_test.go

```diff
@@ -86,22 +86,30 @@ func TestSDAMErrorHandling(t *testing.T) {
 				assert.True(mt, mongo.IsTimeout(err), "expected timeout error, got %v", err)
 				assert.True(mt, mongo.IsNetworkError(err), "expected network error, got %v", err)
 				// Assert that the pool is cleared within 2 seconds.
-				assert.Soon(mt, func(ctx context.Context) {
+				callback := func() bool {
 					ticker := time.NewTicker(100 * time.Millisecond)
 					defer ticker.Stop()
 
 					for {
```

Comment:
As with the tests in client_test.go, this for loop does nothing and we should remove it in favor of the assert.Eventually tick logic.